### PR TITLE
feat(atenlib): add ops(avg pool2d)

### PR DESCRIPTION
**Pull request by xiaowuhu**
Have to use the trace_only = True because of complicated parameters input.

### CODE REVIEW

File: onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py [hunk 1]
```diff
@@ -676,6 +677,11 @@ def _where_input_wrangler(
         reason="fixme: ORT fails with invalid model: 'ONNX Schema aten_adaptive_avg_pool3d: failed validating the check: !(it.GetName().empty())'",
         test_class_name="TestOutputConsistencyFullGraph",
     ),
+    xfail(
+        "nn.functional.avg_pool2d",
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py:681]: is this needed since it’s skipped below already?

File: onnxscript/function_libs/torch_aten/ops/nn.py [hunk 2]
```diff
@@ -125,18 +125,62 @@ def aten_adaptive_max_pool3d_backward(
...
 
+@torch_op("aten::avg_pool2d", trace_only=True)
 def aten_avg_pool2d(
-    self: TensorType,
+    self: TFloat,
     kernel_size: Sequence[int],
-    stride: Optional[Sequence[int]] = None,
-    padding: Sequence[int] = (0, 0),
+    stride: Sequence[int] = None,
+    padding: Sequence[int] = 0,
```
Comment (titaiwangms) [inline onnxscript/function_libs/torch_aten/ops/nn.py:133]:
```suggestion
    padding: Sequence[int] = (0,),
```

File: onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py [hunk 3]
```diff
@@ -682,6 +683,11 @@ def _where_input_wrangler(
...
+        reason="fixme: graph_building.py assert not isinstance(value, TorchScriptTensor)",
+        test_class_name="TestOutputConsistencyFullGraph",
+    ),
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_aten/ops_correctness_test.py:690]:
```suggestion
```

Removing to see what happens. This need to work.

File: onnxscript/function_libs/torch_aten/ops/nn.py [hunk 4]
```diff
@@ -125,18 +125,62 @@ def aten_adaptive_max_pool3d_backward(
     raise NotImplementedError()
 
 
+@torch_op("aten::avg_pool2d", trace_only=True)
 def aten_avg_pool2d(
-    self: TensorType,
+    self: TFloat,
     kernel_size: Sequence[int],
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_aten/ops/nn.py:131]: We may need to use a wrangler to change the np arrays back to a list, if `kernel_size`, `stride` and `padding` are supplied as np.arrays. I suspect this was why there was an assertion error in the full graph test - we passed them as inputs mistakenly because they were preprocessed as np arrays (since they are supplied as positional arguments).

We will soon be able to distinguish between inputs and arguments even better after the OpSchema PR (https://github.com/onnx/onnx/pull/5020) in onnx is merged.

**Review by justinchuby — APPROVED** [review]
